src/Itinerary.hpp

```diff
@@ -2,30 +2,76 @@
 #ifndef Itinerary_hpp
 #define Itinerary_hpp
 
+#include "SparseMatrix.hpp"
+
 #include <concepts>
 #include <utility>
 
 namespace dmf {
 
   template <typename Id>
-    requires std::unsigned_integral<Id>
+  requires std::unsigned_integral<Id>
   class Itinerary {
   private:
     std::pair<Id, Id> m_trip;
+    SparseMatrix<Id, bool> m_path;
 
   public:
     Itinerary() = default;
-    Itinerary(Id source, Id destination) : m_trip{std::make_pair(source, destination)} {}
+    Itinerary(Id source, Id destination);
     explicit Itinerary(std::pair<Id, Id> trip) : m_trip{std::move(trip)} {}
+    Itinerary(Id source, Id destination, SparseMatrix<Id, bool> path);
+    Itinerary(std::pair<Id, Id> trip, SparseMatrix<Id, bool> path);
 
     // Setters
-    void set_source(Id source) { m_trip.first = source; }
-    void set_destination(Id destination) { m_trip.second = destination; }
+    void setSource(Id source);
+    void setDestination(Id destination);
+    void setPath(SparseMatrix<Id, bool> path);
 
     // Getters
-    Id source() const { return m_trip.first; }
-    Id destination() const { return m_trip.second; }
+    Id source() const;
+    Id destination() const;
+    std::pair<Id, Id> trip() const;
+    SparseMatrix<Id, bool> path() const;
```

Comment:
Same question

Reply:
It can be indeed useful for `SparseMatrix`